test/integration/dataplane_mtls_test.go

```diff
@@ -0,0 +1,118 @@
+//go:build dataplane_tests
```

Comment:
I'm not going to say no to adding Golang tests here because this PR review has gone on long enough already and I appreciate your efforts and would prefer to get this in and focus on iterating than to try and stop and do a major rework again now. That said, I felt it would be remiss of me not to point out that we'll likely need to rewrite this into Rust tests at some point so just keep that in mind! Feel free to consider this comment resolved at your discretion.

Reply:
Yeah, I'm aware. I wrote these in go out of laziness; I drew on the precedent of integration tests written in golang.